Components/CategoryExtractor.php

```diff
@@ -95,6 +98,38 @@ public function extractImportedCategories()
     }
 
     /**
+     * @param $parentId
+     * @return array
+     */
+    public function getCategoryIdsCollection($parentId)
+    {
+        $categoryRepository = $this->categoryResolver->getCategoryRepository();
```

Comment:
```getCategoryRepository``` method is not defined in CategoryResolver interface. I think it's not good idea to define method like this in interface. I'd suggest to inject CategoryRepository in CategoryExtractor.

Btw please check ConnectFactory on line 137, when we use ```AutoCategoryResolver``` and when ```DefaultCategoryResolver```.  It depends on createCategoriesAutomatically configuration.